src/redux/books/books.js

```diff
@@ -1,29 +1,72 @@
+import { API_URL, HEADERS } from '../../data/api';
+
 const ADD_BOOK = 'bookstore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
+const GET_BOOK = 'bookstore/books/GET_BOOK';
 
 const initialState = [];
 
-export const addBook = (booksload) => ({
-  type: ADD_BOOK,
-  booksload,
-});
+export const addBook = (book) => async (dispatch) => {
+  await fetch(API_URL, {
+    method: 'POST',
+    headers: HEADERS,
+    body: JSON.stringify(book),
+  })
+    .then((response) => response.text())
+    .then(
+      () => dispatch({ type: ADD_BOOK, payload: book }),
+      () => dispatch({ type: ADD_BOOK, payload: null }),
+    );
+};
+
+export const removeBook = (bookId) => async (dispatch) => {
+  await fetch(`${API_URL}/${bookId}`, {
+    method: 'DELETE',
+    headers: HEADERS,
+    body: JSON.stringify({ item_id: bookId }),
+  })
+    .then((response) => response.text())
+    .then(
+      () => dispatch({ type: REMOVE_BOOK, payload: bookId }),
+      () => dispatch({ type: REMOVE_BOOK, payload: null }),
+    );
+};
```

Comment:
- [ ] Hi @cynthiainga, please don't forget that even after refactoring your code to use the API, the Add book and Remove book features should work as before. I can't remove a particular book. When I click on the remove book button, all the books are removed. Please let's fix this issue.

Kindly make sure that each book has a unique ID

![Screenshot from 2022-03-31 11-16-50](https://user-images.githubusercontent.com/63087491/161034452-36b5be96-2a75-4f29-8fbe-207b72959c99.png)

Reply:
Thanks for reviewing my code, I made the change now the user can delete only one book, I fixed the error.